src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -255,12 +252,13 @@ rpmostreed_deployment_generate_variant (OstreeSysroot *sysroot,
 
       g_variant_dict_insert (&dict, "base-checksum", "s", base_checksum);
       variant_add_commit_details (&dict, "base-", base_commit);
+      /* for layered commits, check if their base commit has end of life attribute */
+      variant_add_metadata_attribute (&dict, "ostree.endoflife", "endoflife", base_commit);
```

Comment:
Can you change this to use `OSTREE_COMMIT_META_KEY_ENDOFLIFE`?  If you want to test locally you'll need to build ostree git master too.